htlcswitch/switch.go

```diff
@@ -856,15 +844,34 @@ func (s *Switch) handleLocalDispatch(pkt *htlcPacket) error {
 // multiple db transactions. The guarantees of the circuit map are stringent
 // enough such that we are able to tolerate reordering of these operations
 // without side effects. The primary operations handled are:
-//  1. Ack settle/fail references, to avoid resending this response internally
-//  2. Teardown the closing circuit in the circuit map
-//  3. Transition the payment status to grounded or completed.
-//  4. Respond to an in-mem pending payment, if it is found.
+//  1. Save the payment result to the pending payment store.
+//  2. Notify subscribers about the payment result.
+//  3. Ack settle/fail references, to avoid resending this response internally
+//  4. Teardown the closing circuit in the circuit map
 //
 // NOTE: This method MUST be spawned as a goroutine.
 func (s *Switch) handleLocalResponse(pkt *htlcPacket) {
 	defer s.wg.Done()
 
+	paymentID := pkt.incomingHTLCID
+
+	// The error reason will be unencypted in case this a local
+	// failure or a converted error.
+	unencrypted := pkt.localFailure || pkt.convertedError
+	n := &networkResult{
+		msg:          pkt.htlc,
+		unencrypted:  unencrypted,
+		isResolution: pkt.isResolution,
+	}
```

Comment:
Add timestamp to record when failure was received. This is one building block of black hole defense

Reply:
Preferably `time.UnixNano` for sub-second resolution